analysers/experiments/constructDuration.py

```diff
@@ -0,0 +1,84 @@
+import sys
+import json
+import io
+import gzip
+import uuid
+import math
+
+from datetime import timedelta
+
+from pyspark_cassandra import CassandraSparkContext
+from pyspark_cassandra import RowFormat
+from pyspark import SparkConf
+
+def createQuery(CassandraRDD, experimentID):
+    from commons import computeExperimentMetrics, computeModeMinMax
+    
+    queries = []
+    
+    combinations = CassandraRDD.map(lambda a: (a["construct_type"], a["construct_name"])).distinct().collect()
+    
+    for combs in combinations:
+        dataRDD = CassandraRDD.filter(lambda r: r['construct_name'] == combs[1] and r['construct_type'] == combs[0])
+        
+        if dataRDD.isEmpty():
+            continue
+        
+        metrics = computeExperimentMetrics(dataRDD, "construct_duration")
+        metrics.update(computeModeMinMax(dataRDD, "construct_duration"))
+        
+        type = combs[0]
+        name = combs[1]
+        
+        if combs[0] is None:
```

Comment:
This check should be at the beginning, what happens if it is `Unspecified` for the computation on https://github.com/benchflow/analysers/pull/75/files#diff-dc119ca3b3977aad5d2b2e2434d76438R22